apps/worker/src/app/shared/utils/should-halt-on-step-failure.ts

```diff
@@ -0,0 +1,27 @@
+import { StepTypeEnum } from '@novu/shared';
+
+export const shouldHaltOnStepFailure = (
+  stepType: StepTypeEnum | undefined,
+  shouldStopOnFail: boolean | undefined
+): boolean | undefined => {
+  if (!stepType) {
+    return shouldStopOnFail;
+  }
+
+  /*
+   * Action steps always stop on failure across all versions (v1 & v2)
+   */
+  if (stepType === StepTypeEnum.DELAY) {
```

Comment:
What about Digest?

Reply:
good catch! 
it should been isActionStep 🤦‍♂️
probably missed it while developing and refactoring this..